### ISSUE TITLE
withSortKeyCondition(beginsWith()) throws error

### ISSUE THREAD

**Issue by henriqueguisasola** [lucasmafra/type-dynamo#5]
When you use the beginsWith expression with the withSortKeyCondition a dynamo error is thrown. I'm using the TypeDynamo version 0.1.3.

Here is my code:

```
const items = await this.repo.onIndex.idAllRoomsKeyIndex
      .find({ id: userId })
      .withSortKeyCondition(beginsWith('all'))
      .execute()
```

And here is the error:
```
{ ValidationException: Invalid KeyConditionExpression: Invalid function name; function: beginsWith,
message: 'Invalid KeyConditionExpression: Invalid function name; function: beginsWith',
code: 'ValidationException',
time: 2018-10-29T19:55:50.196Z,
requestId: 'PKM2UTJ3ERP495736KR0SJO4PJVV4KQNSO5AEMVJF66Q9ASUAAJG',
statusCode: 400,
retryable: false,
retryDelay: 35.338869028594075 }
```

That is probably happening because the correct dynamo sintax is `begins_with()` instead of `beginsWith()`

**Comment by henriqueguisasola** [comment]
I was looking at the code, trying to find the problem, and i found this in the `src/helpers/key-condition-expression-generator` file:

```
  // TODO withSortKeyCondition
  public generateExpression(
    partitionKey: object, withSortKeyCondition?: IWithSortKeyCondition,
  ) {
    const keyName = this.getKeyNameFrom(partitionKey)
    const keyValue = this.getKeyValueFrom(partitionKey, keyName)

    const expressionAttributeNames = this.expressionAttributeNamesGenerator
        .generateExpression([keyName])

    const expressionAttributeValues = this.expressionAttributeValuesGenerator
        .generateExpression([keyValue])

    const keyConditionExpression = this.generateKeyConditionFrom(
      expressionAttributeNames, expressionAttributeValues,
    )

    return {
      expressionAttributeNames,
      expressionAttributeValues,
      keyConditionExpression,
    }
  }
```

Is this functionality not implemented yet? Or am I just looking at the wrong place?

**Comment by henriqueguisasola** [comment]
So, i went into my node_modules folder, and found this piece of code in the `type-dynamo/dist/expressions/index.js` file:

```
function beginsWith(subString) {
    const randomId = ':' + random_generator_1.randomGenerator();
    return {
        value: 'beginsWith',
        functionOperand: randomId,
        type: 'function',
        expressionAttributeValues: {
            [randomId]: {
                ['S']: subString,
            },
        },
        kind: 'beginsWith',
    };
}
exports.beginsWith = beginsWith;
```

If this is the code that adds the beginsWith operator in the KeyConditionExpression, it should be `value: begins_with`, or something like this.

DynamoDB reference: https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Query.html#Query.KeyConditionExpressions

**Comment by lucasmafra** [comment]
Hi @henriqueguisasola 

Thanks for reporting this bug. You're absolutely right, it should be 'begins_with' instead of 'beginsWith'. Already fixed [here](https://github.com/lucasmafra/type-dynamo/commit/540213c53b9b0d2f5b8a6f4a9e6443e412b4eaac) and available on [v0.1.4](https://www.npmjs.com/package/type-dynamo)
